qa/abstracts/src/main/java/io/camunda/zeebe/process/test/qa/abstracts/assertions/AbstractMessageAssertTest.java

```diff
@@ -308,6 +309,9 @@ void testHasNotExpiredFailure() throws InterruptedException, TimeoutException {
               Collections.emptyMap());
       Utilities.increaseTime(engine, timeToLive.plusMinutes(1));
 
+      // #960 logging current stream of events for assert troubleshooting
+      RecordStream.of(engine.getRecordStreamSource()).print(true);
```

Comment:
❌ Let's surround the assertion with try catch and log the record stream only if it fails. We can can re-throw the assertion error.

Please also consider adding a description that the code will be removed once the flakiness is resolved while adding the issue link.

Reply:
Thank you, great suggestion, done 👍🏻